### ISSUE TITLE
add support for scipy

### ISSUE THREAD

**Issue by ktotam1** [mhostetter/galois#514]
it would be nice to be able to use certain methods of scipy, such as scipy.linalg.null_space, which numpy does not have.

**Comment by mhostetter** [comment]
I agree this would be nice. However, currently SciPy doesn't support subclasses of `ndarray`. Also, there are no hooks in `np.ndarray` that can catch and override SciPy function calls. If you know of a way, I'm very open to the idea.

For the specific feature you mentioned, maybe consider this method https://mhostetter.github.io/galois/latest/api/galois.FieldArray.null_space/.
